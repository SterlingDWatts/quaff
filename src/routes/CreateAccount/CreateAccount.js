import React, { Component } from "react";
import CreateAccountForm from "../../components/CreateAccountForm/CreateAccountForm";
import "./CreateAccount.css";

class CreateAccount extends Component {
  static defaultProps = {
    history: {
      push: () => {},
    },
  };

  handleRegistrationSuccess = () => {
    this.props.history.push("/login");
  };

  render() {
    return (
      <div className="CreateAccount">
        <CreateAccountForm
          onRegistrationSuccess={this.handleRegistrationSuccess}
        />
      </div>
    );
  }
}

export default CreateAccount;
