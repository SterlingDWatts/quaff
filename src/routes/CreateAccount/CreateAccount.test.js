import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import CreateAccount from "./CreateAccount";

describe("CreateAccount Component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <CreateAccount />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <CreateAccount />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
