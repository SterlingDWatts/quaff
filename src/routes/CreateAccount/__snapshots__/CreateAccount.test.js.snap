// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccount Component renders the UI as expected 1`] = `
<div
  className="CreateAccount"
>
  <form
    className="CreateAccountForm"
    onSubmit={[Function]}
  >
    <h2>
      Create Account
    </h2>
    <div
      role="alert"
    />
    <div
      className="hint"
    >
      <span
        className="Required "
      >
        *
      </span>
       required fields
    </div>
    <div
      className="LabelGroup CreateAccountForm__first_name"
    >
      <label
        className="Label "
        htmlFor="firstName"
      >
        First Name 
        <span
          className="Required "
        >
          *
        </span>
      </label>
      <input
        className="Input "
        id="firstName"
        name="firstName"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <div
        className="NoError"
      >
         
      </div>
    </div>
    <div
      className="LabelGroup CreateAccountForm__last_name"
    >
      <label
        className="Label "
        htmlFor="lastName"
      >
        Last Name 
        <span
          className="Required "
        >
          *
        </span>
      </label>
      <input
        className="Input "
        id="lastName"
        name="lastName"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <div
        className="NoError"
      >
         
      </div>
    </div>
    <div
      className="LabelGroup CreateAccountForm__username"
    >
      <label
        className="Label "
        htmlFor="username"
      >
        Username 
        <span
          className="Required "
        >
          *
        </span>
      </label>
      <input
        className="Input "
        id="username"
        name="username"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <div
        className="NoError"
      >
         
      </div>
    </div>
    <div
      className="LabelGroup CreateAccountForm__email"
    >
      <label
        className="Label "
        htmlFor="email"
      >
        Email 
        <span
          className="Required "
        >
          *
        </span>
      </label>
      <input
        className="Input "
        id="email"
        name="email"
        onChange={[Function]}
        required={true}
        type="email"
      />
      <div
        className="NoError"
      >
         
      </div>
    </div>
    <div
      className="LabelGroup CreateAccountForm__confirm_email"
    >
      <label
        className="Label "
        htmlFor="confirmEmail"
      >
        Confirm Email 
        <span
          className="Required "
        >
          *
        </span>
      </label>
      <input
        className="Input "
        id="confirmEmail"
        name="confirmEmail"
        onChange={[Function]}
        required={true}
        type="email"
      />
      <div
        className="NoError"
      >
         
      </div>
    </div>
    <div
      className="LabelGroup CreateAccountForm__password"
    >
      <label
        className="Label "
        htmlFor="password"
      >
        Password 
        <span
          className="Required "
        >
          *
        </span>
      </label>
      <input
        className="Input "
        id="password"
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
      />
      <div
        className="password-requirements"
      >
        <p>
          <span
            className="Required "
          >
            *
          </span>
           Password required
        </p>
        <p>
          <span
            className="Required "
          >
            *
          </span>
           Cannot contain a space
        </p>
        <p>
          <span
            className="Required "
          >
            *
          </span>
           Requires a lower case character
        </p>
        <p>
          <span
            className="Required "
          >
            *
          </span>
           Requires an upper case character
        </p>
        <p>
          <span
            className="Required "
          >
            *
          </span>
           Requires a number
        </p>
        <p>
          <span
            className="Required "
          >
            *
          </span>
           Requires a special character
        </p>
        <p>
          <span
            className="Required "
          >
            *
          </span>
           Requires 8-72 characters
        </p>
      </div>
    </div>
    <div
      className="LabelGroup CreateAccountForm__confirm_password"
    >
      <label
        className="Label "
        htmlFor="confirmPassword"
      >
        Confirm Password 
        <span
          className="Required "
        >
          *
        </span>
      </label>
      <input
        className="Input "
        id="confirmPassword"
        name="confirmPassword"
        onChange={[Function]}
        required={true}
        type="password"
      />
      <div
        className="NoError"
      >
         
      </div>
    </div>
    <div
      className="CreateAccountForm--error"
    >
      <div
        className="NoError"
      >
         
      </div>
    </div>
    <div
      className="CreateAccountForm__buttons"
    >
      <button
        className="Button form__button"
        disabled="Requires only letters, -, and '"
        type="submit"
      >
        Create Account
      </button>
    </div>
    <div
      className="CreateAccountForm__other_links"
    >
      Already have an account? 
      <a
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </div>
  </form>
</div>
`;
